Program.cs

```diff
@@ -200,6 +236,97 @@ static void Main(string[] args)
             }
         }
 
+        private static int GetPreviousMetadataEpoch(
+            string outputDirectory,
+            string epochType)
+        {
+            try
+            {
+                string previousMetadata = Directory.GetFiles(outputDirectory)
+                    .Where(x => Path.GetExtension(x) == ".json")
+                    .OrderByDescending(x => File.GetLastWriteTime(x))
+                    .First();
+                using (StreamReader reader = new StreamReader(previousMetadata))
+                {
+                    var jsonString = reader.ReadToEnd();
+                    var jsonObject = JObject.Parse(jsonString); 
+                    return (int)jsonObject[epochType];            
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+                return 0;
+            }
+        }
+        private static Block<T> GetLastestBlockWithTransaction<T>(Block<T> tip, RocksDBStore store)
```

Comment:
IMO this seems better to read.
```suggestion
        }

        private static Block<T> GetLastestBlockWithTransaction<T>(Block<T> tip, RocksDBStore store)
```